### ISSUE TITLE
JSHint-like overrides

### ISSUE THREAD

**Issue by jeffm** [philbooth/complexity-report#39]
In jshint, I can define a list of global style standards. For specific cases, I can include comments that tell jshint to ignore one or more specific warnings. So, for example, /* -W040 */ before a function, and /* +W040 */ after a function, will cause jshint to ignore the 'strict violation' warning within that function. 

In complexity-report, I can define a list of global thresholds for various metrics. For the exact same reasons that overriding style warnings is important in jshint, and given that code complexity is far more subjective and difficult than where brackets should go, I would like to be able to suppress warnings if I decide that a specific function is not too complex for my situation. In other words, I'd like to be able to override the global values for maintainability, cyclomatic complexity, etc. at the function and module level.

With this capability, if I integrate complexity checks into my BDD/TDD & CI/CD processes, I don't have to set excessively high global thresholds just because I have one or two miscreant functions. This would give my team the ability to think through what the metrics + code are saying, make reasoned choices as to what is acceptable complexity with high specificity, and more effectively manage complexity.

**Comment by philbooth** [comment]
Yep, good shout, I agree. I'm away on hols right now but will take a stab
at something when I get back.

**Comment by philbooth** [comment]
Unfortunately I've stopped maintaining complexity-report, so this issue will not get addressed by me. I'm not actively using the project myself any more and my attention is currently elsewhere so I can't give it the time it requires.

If anyone out there is interested in taking the project forward, they should definitely get in touch with me so we can chat and sort it out.

I'm adding this note to all open issues and have added a similar note to the readme, just to let people know. Sorry for the inconvenience. :)
